lms/djangoapps/instructor/views/api.py

```diff
@@ -1851,23 +1922,92 @@ def list_email_content(request, course_id):
     return JsonResponse(response_payload)
 
 
+@view_auth_classes()
+class InstructorTasks(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Lists currently running instructor tasks
+
+    **Parameters**
+       - With no arguments, lists running tasks.
+       - `problem_location_str` lists task history for problem
+       - `problem_location_str` and `unique_student_identifier` lists task
+           history for problem AND student (intersection)
+
+    **Example Requests**:
+
+        GET /courses/{course_id}/instructor/api/v0/tasks
+
+    **Response Values**
+        {
+          "tasks": [
+            {
+              "status": "Incomplete",
+              "task_type": "grade_problems",
+              "task_id": "2519ff31-22d9-4a62-91e2-55495895b355",
+              "created": "2019-01-15T18:00:15.902470+00:00",
+              "task_input": "{}",
+              "duration_sec": "unknown",
+              "task_message": "No status information available",
+              "requester": "staff",
+              "task_state": "PROGRESS"
+            }
+          ]
+        }
+    """
+    @apidocs.schema(parameters=[
+        apidocs.string_parameter(
+            'course_id',
+            apidocs.ParameterLocation.PATH,
+            description="ID for the course whose tasks need to be listed.",
+        ),
+        apidocs.string_parameter(
+            'problem_location_str',
+            apidocs.ParameterLocation.QUERY,
+            description="Filter instructor tasks to this problem location.",
+        ),
+        apidocs.string_parameter(
+            'unique_student_identifier',
+            apidocs.ParameterLocation.QUERY,
+            description="Filter tasks to a singe problem and a single student. "
+                        "Must be used in combination with `problem_location_str`.",
+        ),
+    ])
+    def get(self, request, course_id):
+        """
+        List instructor tasks filtered by `course_id`.
+        """
+        return _list_instructor_tasks(request=request, course_id=course_id)
+
+
 @require_POST
 @ensure_csrf_cookie
-@cache_control(no_cache=True, no_store=True, must_revalidate=True)
-@require_course_permission(permissions.SHOW_TASKS)
 def list_instructor_tasks(request, course_id):
     """
     List instructor tasks.
 
-    Takes optional query paremeters.
+    Takes optional query parameters.
         - With no arguments, lists running tasks.
         - `problem_location_str` lists task history for problem
         - `problem_location_str` and `unique_student_identifier` lists task
             history for problem AND student (intersection)
     """
+    return _list_instructor_tasks(request=request, course_id=course_id)
+
+
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_course_permission(permissions.SHOW_TASKS)
+def _list_instructor_tasks(request, course_id):
```

Comment:
The original API uses POST, while the new endpoint is GET, so I've created a very small wrapper to just add the post requirement and CSRF requirement.